utilities/src/main/java/ru/study21/jcsv/xxl/analyzer/CSVCustomizableAnalyzer.java

```diff
@@ -146,15 +162,88 @@ public void acceptRow(List<String> row) {
                     maxValuesIntAction(colIndex, 1);
 
             @Override
-            public void acceptRow(List<String> row) {
+            public void acceptRow(CSVRow row) {
                 delegate.acceptRow(row);
             }
 
             @Override
             public Integer getResult() {
                 return delegate.getResult().get(0);
             }
+
+            @Override
+            public void finish() {}
+        };
+    }
+
+    public static Action<Long> withOffsetsSizeAction(int colIndex) {
+        return new Action<>() {
+            Long size = 0L;
+
+            @Override
+            public void acceptRow(CSVRow row) {
+                size += Long.SIZE + row.get(colIndex).getBytes(StandardCharsets.UTF_8).length;
```

Comment:
We must be aware that this size:

* does not include field separators.
* sum of all such sizes doesn't take into account new lines chars
* this value does not help us to handle possible absense of newline in last row
* there are several ways of escaping inside rows (https://tools.ietf.org/html/rfc4180) - and  CSVRow contains data after some processing of these escaping rules.

So it is just rough estimate